### PR TITLE
Better training on all models

### PR DESCRIPTION
**Pull request by bcipolli**
Training has been a bit incomplete. Here, I try to make things more robust by:
* Validating (and fixing) various dataset issues (vanhateren selects really bad patches sometimes; kitterle images are not normalized well and square waves don't have sharp boundaries, cycles/image are not selected carefully, etc.
* Getting tough-to-train datasets working (i.e. kitterle and christman)

### CODE REVIEW

**Comment by bcipolli** [comment]
The Kitterle model is overfitting on the square/sin wave discrimination. Neither network learns it well. You can see that ordering of models flips on train vs. test set errors.

Train (look at the right discrimination):
uber_kitterle u0 p0 _106176406-train-linear-r20-sf_mixed-recog_freq-small-cycles-2 5

Test (look at the right discrimination)
uber_kitterle u0 p0 _106176406-test-linear-r20-sf_mixed-recog_freq-small-cycles-2 5

Note that the two frequencies for the frequency difference should be to the left of the frequency difference curve, as it shows a RH dominance. 

uber_kitterle u0 p0 _44025637-fft-1d-log-diff-pct-smoothed-0 50-r20-250--small
